packages/ui/src/machines/authenticator/actors/resetPassword.ts

```diff
@@ -100,6 +104,39 @@ export function resetPasswordActor({ services }: ResetPasswordMachineOptions) {
                 },
               },
             },
+            sendConfirmPassword: {
+              tags: ['pending'],
+              entry: ['sendUpdate', 'clearError'],
+              initial: 'init',
+              states: {
+                init: {
+                  always: [
+                    {
+                      target: 'pending',
+                      cond: 'shouldAutoConfirmReset',
+                    },
+                    {
+                      target: 'valid',
+                    },
+                  ],
+                },
```

Comment:
This always runs since `confirmResetPassword` is set to `parallel` . However it will skip sending the reset password if intent is not sent.

Reply:
Does running an extraneous action seems return modified state? cc @wlee221

Reply:
It will update the state long as long as `sendUpdate` is there. 

That said, I think that this does not need to be its own state since what we're adding is more of a side effect that just runs once initially. So I think just adding action can be  sufficient. 

What do you think about adding an action:

```
actions: {
  ...
  resendCode: async (context, event) => {
    if (context.intent === 'confirmPasswordReset')
      await Auth.forgotPassword(...)
    }
  }
}
```

and then calling that when we first enter `confirmResetPassword` state?

Reply:
I wasn't sure if it was ok, to add this as an action instead, I'll try that out though!~

Reply:
[98dc3ff](https://github.com/aws-amplify/amplify-ui/pull/3201/commits/98dc3ff23b87b70bd18fb351d9cac58cadae8b61)